### PR TITLE
User can sign in to the website

### PR DESCRIPTION
**Pull request by Fraseer**
PT Link: https://www.pivotaltracker.com/story/show/179905126

## User Story
As a registered user
In order to see restricted articles
I would like to sign into my account

## Changes Proposed
- Create Sign In Page
- Add sign in button to header
- POST request for sign in
- Stub out two calls for cypress
- Include auth token in headers

## Screenshots
0qdNvdBVXb

image
image

### CODE REVIEW

**Review by Fraseer — COMMENTED** [review]
Please can someone add the Swedish translations :)

File: cypress.json [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
-  "baseUrl": "http://localhost:3001",
-  "chromeWebSecurity": false
+  "baseUrl": "http://localhost:3000",
+  "chromeWebSecurity": false,
+  "theme": "dark",
```
Comment (maxarvid) [inline cypress.json:4]: I'm guessing we should not assume every user has a dark theme on their browser

Comment (Fraseer) [reply cypress.json:4]: this is dark theme on cypress

File: cypress/integration/userSeesDifferentCategoryMenus.feature.js [hunk 2]
```diff
@@ -8,14 +8,18 @@ describe("The Categories render differently on mobile and desktop", () => {
...
+        cy.get("[data-cy=mobile-category-list]").click().children().should("have.length", 3);
+        cy.get("[data-cy=sign-up-button]").should("be.visible")
+        cy.get("[data-cy=sign-in-button]").should("be.visible")
+      });
     });
   });
   describe("when the user is on a desktop", () => {
     it("is expected that the user sees a subheader with the news categories listed", () => {
-      cy.get("[data-cy=category-list]").should('be.visible')
+      cy.get("[data-cy=category-list]").should("be.visible");
```
Comment (AugustEnhager) [inline cypress/integration/userSeesDifferentCategoryMenus.feature.js:22]: whoopsie

File: src/components/SignIn.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              closeOnClick
+              rtl={false}
+              pauseOnFocusLoss
+              draggable
+              pauseOnHover
+            />
+          </div>
+        </Container>
+      )}
+    </>
```
Comment (AugustEnhager) [inline src/components/SignIn.jsx]: This empty tag is needed because the ternary is fragile right?

Comment (Fraseer) [reply src/components/SignIn.jsx]: Yeah that one is needed

**Review by AugustEnhager — APPROVED** [review]

File: cypress.json (hunk 1 above)
Comment (maxarvid) [reply cypress.json:4]: I learned something new today!

**Review by maxarvid — APPROVED** [review]
The toasters look awesome

File: cypress/support/index.js [hunk 4]
```diff
@@ -1 +1,2 @@
 import "./commands";
+require('cypress-dark')
```
Comment (tochman) [inline cypress/support/index.js]: Why require when the other line uses import?

Comment (Fraseer) [reply cypress/support/index.js]: Thats what it said on the documentation, are they interchangeable?

File: src/components/SignIn.jsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import "react-toastify/dist/ReactToastify.css";
+import auth from "../modules/auth";
+import { useDispatch } from "react-redux";
+import { Redirect } from "react-router-dom";
+
+const SignIn = () => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const [redirect, setRedirect] = useState(false);
+  const delay = require("delay");
```
Comment (tochman) [inline src/components/SignIn.jsx]: Another require? I don’t understand.

**Pull request merged**